src/features/impl_rkyv.rs

```diff
@@ -0,0 +1,197 @@
+#[cfg(feature = "bytecheck")]
+macro_rules! impl_rkyv {
+    (@bytecheck $type:ty) => {
+        impl<C: ?Sized> bytecheck::CheckBytes<C> for $type {
+            type Error = core::convert::Infallible;
+
+            #[inline]
+            unsafe fn check_bytes<'a>(
+                value: *const Self,
+                _: &mut C,
+            ) -> Result<&'a Self, Self::Error> {
+                Ok(&*value)
+            }
+        }
+    };
+
+    ($type:ty) => {
+        impl_rkyv_derive!(@serialize $type);
+        impl_rkyv_derive!(@archive_deserialize $type);
+        impl_rkyv!(@bytecheck $type);
+    };
+}
+
+#[cfg(not(feature = "bytecheck"))]
+macro_rules! impl_rkyv {
+    ($type:ty) => {
+        impl_rkyv_derive!(@serialize $type);
+        impl_rkyv_derive!(@archive_deserialize $type);
+    };
+}
+
+macro_rules! impl_rkyv_derive {
+    (@serialize $type:ty) => {
+        impl<S: Fallible + ?Sized> Serialize<S> for $type {
+            #[inline]
+            fn serialize(&self, _: &mut S) -> Result<Self::Resolver, S::Error> {
+                Ok(())
+            }
+        }
+    };
+
+    (@archive_deserialize $type:ty) => {
+        const _: () = {
+            #[cfg(not(any(feature = "archive_le", feature = "archive_be")))]
+            type Archived = $type;
+            #[cfg(feature = "archive_le")]
+            type Archived = rend::LittleEndian<$type>;
```

Comment:
I'm not sure if this will work right. There might be some work to do here if you want glam types to support little- and big-endian, since some of the SIMD types are new primitives. If glam is using a standard crate I can see about getting this work done.

Reply:
I removed the endian conversion step for now until support has been added to convert SIMD types.

Reply:
Awesome, looks great! I'll be digging into this more comprehensively in https://github.com/rkyv/rend/issues/4